converter/src/main/java/gov/cms/qpp/conversion/decode/QrdaDecoderEngine.java

```diff
@@ -165,6 +169,33 @@ private DecodeData decodeChildren(final Element element, final Node parentNode)
 		return decodeData;
 	}
 
+	/**
+	 * Reduces the {@code templateId} {@link Element}s so there are no duplicates.  All other {@link Element}s are left alone.
+	 *
+	 * @param childElements The elements to filter
+	 * @return A {@link List} of {@link Element}s that are filtered.
+	 */
+	private List<Element> getUniqueTemplateIdElements(final List<Element> childElements) {
+		Set<TemplateId> uniqueTemplates = new HashSet<>();
+
+		return childElements.stream()
+			.filter(filterElement -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)